src/features/callAssignments/models/CallAssignmentModel.ts

```diff
@@ -93,6 +95,35 @@ export default class CallAssignmentModel {
     return data.target.filter_spec?.length === 0;
   }
 
+  setCooldown(cooldown: number) {
+    const state = this._store.getState();
+    const callAssignment = state.callAssignments.callAssignments.find(
+      (ca) => ca.id == this._id
+    );
+
+    if (callAssignment) {
+      this._store.dispatch(callAssignmentLoad());
+      const promise = fetch(
+        `/api/orgs/${this._orgId}/call_assignments/${this._id}`,
+        {
+          body: JSON.stringify({ cooldown }),
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          method: 'PATCH',
+        }
+      )
+        .then((res) => {
+          return res.json();
+        })
+        .then((data: { data: CallAssignmentData }) => {
+          this._store.dispatch(callAssignmentUpdated(data.data));
+        });
+
+      throw promise;
```

Comment:
I don't think this is used anywhere or required here?

Reply:
the throw promise line?

Reply:
Yeah, is there actually anything catching it (a `<Suspense/>`), or is it printed to the console as an uncaught error?

Reply:
No, i just copied from the pattern with "throw promise" that you used when you sketched/made the first getter.